src/components/30-organisms/footer/index.js

```diff
@@ -68,35 +62,59 @@ class AXAFooter extends InlineStyles {
     this.inlineStyles('resetHeadingCss');
   }
 
-  prepareSlotsWithIndexes() {
-    // set slotted numbers automatically for all titles and all items for each column
-    // every title followed with items will be a column (DOM sequence defines order)
-    // There is no wrapper for the columns but a counter that controls which
-    // column to use
-    // EXAMPLE IN PSEUDO DOM:
-    //
-    // TITLE* -> start column 0
-    // ITEM -> add item to column 0
-    // ITEM -> add item to column 0
-    // ITEM -> add item to column 0
-    // ITEM -> add item to column 0
-    //
-    // TITLE* -> start column 1
-    // ITEM -> add item to column 1
-    // ITEM -> add item to column 1
-    // ITEM -> add item to column 1
-
-    const childrenArr = Array.prototype.slice.call(this.children);
+  _revealChildrenWithSlotAttribute(rawNode, slotElements) {
+    if (typeof rawNode.hasAttribute === 'function') {
+      if (rawNode.hasAttribute('slot')) {
+        slotElements.push(rawNode);
+      }
+      if (rawNode.hasChildNodes()) {
+        const children = Array.prototype.slice.call(rawNode.childNodes);
+        children.forEach(ch => {
+          this._revealChildrenWithSlotAttribute(ch, slotElements);
+        });
+      }
+    }
+  }
+
+  /**
+   * Takes all the child elements and extracts all the ones that come with a
+   * slot-attribute. Those elements will then be changed to be distinguishable
+   * from each other. Each block of links will need a title, so that the block will be displayed.
+   *
+   * Each title marks the beginning of a new column:
+   *
+   * TITLE -> start column 0
+   * ITEM -> add item to column 0
+   * ITEM -> add item to column 0
+   * ITEM -> add item to column 0
+   * ITEM -> add item to column 0
+   *
+   * TITLE* -> start column 1
+   * ITEM -> add item to column 1
+   * ITEM -> add item to column 1
+   * ITEM -> add item to column 1
+   */
+  _prepareSlotsWithIndexes() {
+    const rawChildren = Array.prototype.slice.call(this.children);
+
+    const allChildrenWithSlotAttribute = [];
+    rawChildren.forEach(rawNode =>
+      this._revealChildrenWithSlotAttribute(
+        rawNode,
+        allChildrenWithSlotAttribute
+      )
+    );
+    this.innerHTML = '';
+    allChildrenWithSlotAttribute.forEach(c => this.appendChild(c));
+
+    const childrenArr = allChildrenWithSlotAttribute;
 
     const filter = criteria => child =>
       child.getAttribute('slot').includes(criteria);
 
-    const noHeaderFilter = criteria => child => {
-      const { nodeName } = child;
-      return (
-        filter(criteria)(child) && !HEADINGS.includes(nodeName.toLowerCase())
-      );
-    };
+    const noHeaderFilter = criteria => child =>
+      filter(criteria)(child) &&
+      !HEADINGS.includes(child.nodeName.toLowerCase());
 
     const onlyColumns = childrenArr.filter(
```

Comment:
use `allChildrenWithSlotAttribute`instead of `childrenArr `and delete row 110

Reply:
ah right, I forgot. this was some leftover from the previous version